# Subject: [patch 059/167] ppc64: override command line AS/LD/CC variables when adding -m64 and co for biarch compilers
# To: torvalds@osdl.org
# Cc: akpm@osdl.org, sven.luther@wanadoo.fr, anton@samba.org,
#    benh@kernel.crashing.org, paulus@samba.org
# From: akpm@osdl.org
#
# The following kind of calls currently fails :
#
#  make ARCH=ppc64 CC="gcc-3.4"
#
# Since the code for detecting a biarch compiler and adding the needed 64bit
# magic argument fails if the AS/LD/CC commands are overriden in the command
# line.
#
# The attached patch fixes this by using the make override and += directive,
# but i am not 100% sure this will work without gmake, as i am no Makefile
# expert.
# 
# Cc: Paul Mackerras <paulus@samba.org>
# Cc: Anton Blanchard <anton@samba.org>
# Cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
# Signed-off-by: Andrew Morton <akpm@osdl.org>
# ---
#
# arch/ppc64/Makefile |    6 +++---
# 1 files changed, 3 insertions(+), 3 deletions(-)
#
diff -puN arch/ppc64/Makefile~ppc64-override-command-line-as-ld-cc-variables-when-adding-m64-and-co-for-biarch-compilers arch/ppc64/Makefile
--- 25/arch/ppc64/Makefile~ppc64-override-command-line-as-ld-cc-variables-when-adding-m64-and-co-for-biarch-compilers	2005-06-21 16:29:10.000000000 -0700
+++ 25-akpm/arch/ppc64/Makefile	2005-06-21 16:29:10.000000000 -0700
@@ -35,9 +35,9 @@ CROSS32AS	:= $(AS) -a32
 CROSS32LD	:= $(LD) -m elf32ppc
 CROSS32OBJCOPY	:= $(OBJCOPY)
 endif
-AS              := $(AS) -a64
-LD              := $(LD) -m elf64ppc
-CC		:= $(CC) -m64
+override AS	+= -a64
+override LD	+= -m elf64ppc
+override CC	+= -m64
 endif
 
 export CROSS32CC CROSS32AS CROSS32LD CROSS32OBJCOPY
_
